packages/proxy/lib/http/util/service-worker-injector.ts

```diff
@@ -203,60 +216,76 @@ export const injectIntoServiceWorker = (body: Buffer) => {
       return oldRemoveEventListener(type, listener, options)
     }
 
-    const originalPropertyDescriptor = Object.getOwnPropertyDescriptor(
+    const originalOnFetchPropertyDescriptor = Object.getOwnPropertyDescriptor(
       self,
       'onfetch',
     )
 
-    if (!originalPropertyDescriptor) {
-      return
+    if (originalOnFetchPropertyDescriptor) {
+      // Overwrite the onfetch property so we can
+      // determine if the service worker handled the request
+      Object.defineProperty(
```

Comment:
do we have to overwrite the object or can we just proxy the values with `Proxy`?

Reply:
Possibly, `self` itself is readonly but I may look into this for a future PR since we are already overwriting multiple other properties/functions in this file.